appstudio-controller/controllers/appstudio.redhat.com/devsandbox_deployment_controller.go

```diff
@@ -0,0 +1,214 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+"github.com/codeready-toolchain/toolchain-common/pkg/condition"
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package appstudioredhatcom
+
+import (
+	"context"
+	"fmt"
+
+	codereadytoolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	applicationv1alpha1 "github.com/redhat-appstudio/application-api/api/v1alpha1"
+	apierr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+const (
+	// deploymentTargetClaimLabel is the label indicating the DeploymentTargetClaim that's associated with the object
+	deploymentTargetClaimLabel = "appstudio.openshift.io/dtc"
+)
+
+// DevsandboxDeploymentReconciler reconciles a SpaceRequest object
+type DevsandboxDeploymentReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=deploymenttargetclaims,verbs=get;list;watch;update;patch
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=deploymenttargetclaims/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=deploymenttarget,verbs=get;list;create;watch;update;patch;delete
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=deploymenttarget/status,verbs=get;update;patch
```

Comment:
Should these objects be plural, such as `resources=deploymenttargets`?

Reply:
The controller will create DeploymentTarget CRs. They are needed here.

Reply:
I mean we should set them like `//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=deploymenttargets,verbs=get;list;create;watch;update;patch;delete`, not `//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=deploymenttarget,verbs=get;list;create;watch;update;patch;delete`, is it?

Reply:
Seems no. "resources" here is plural also.